janus_core/calculations/md.py

```diff
@@ -1120,31 +1123,47 @@ def _run_dynamics(self) -> None:
                 self.temp_start + ramp_sign * i * self.temp_step for i in range(n_temps)
             ]
 
-            if self.logger:
-                self.logger.info("Beginning temperature ramp at %sK", temps[0])
-            if self.tracker:
-                self.tracker.start_task("Temperature ramp")
+            if self.restart:
+                ramp_steps_completed = self.offset // heating_steps
+                ramp_steps_completed = min(ramp_steps_completed, len(temps))
+                if np.isclose(self.temp_start, 0.0):
+                    # T~0K steps are skipped, so offset will not include them.
+                    ramp_steps_completed += 1
+                temps = temps[ramp_steps_completed:]
 
-            for temp in temps:
-                self.temp = temp
-                self._set_velocity_distribution()
-                if isclose(temp, 0.0):
-                    # Calculate forces and energies to be output
-                    self.struct.get_potential_energy()
-                    self.struct.get_forces()
+            if len(temps) > 0:
+                if self.logger:
+                    self.logger.info("Beginning temperature ramp at %sK", temps[0])
+                if self.tracker:
+                    self.tracker.start_task("Temperature ramp")
+
+                for temp in temps:
+                    self.temp = temp
+                    steps = heating_steps
+                    if temp == temps[0]:
+                        steps -= self.offset % heating_steps
```

Comment:
Is this supposed to be `if first step`? Comparison against `0`th element seems risky and masks that.

Reply:
I think it was literally checking what it says - are we on the first temperature of the (truncated) list we're looping through? What's the risk?